app/furniture/marketplace/marketplace.rb

```diff
@@ -122,10 +122,8 @@ def square_connection
       end
     end
 
-    private
-
     def square_order_notifications_enabled?
-      square_location_id.present?
+      square_location_id.present? and square_access_token.present?
```

Comment:
Bear in mind that `&&` and `and` have different operator precedence; so if `square_location_id.present?` is `false` and `square_access_token.present?` is `true`; I think this will return true.

Reply:
Got my operators mixed, didn't even realize ruby supported `and`.

Reply:
Ah, all the more reason for an isolated test of this method then!